File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/LoanInitRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor @RequiredArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class LoanInitRequest  extends SignBase {
    @NonNull
    private String merchantId;
    @NonNull
    private String orderNo;
    @NonNull
    private String purchaseId;
    @NonNull
    private String clientIp;
    @NonNull
    private Long totalAmount;
    @NonNull
    private String currency;
    @NonNull
    private String returnUrl;
    @NonNull
    private String returnMethod;
    private LoanCustomer customer;
    private String merchantData;
    @NonNull
    private String language;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, orderNo);
        add(sb, purchaseId);
        add(sb, dttm);
        add(sb, clientIp);
        add(sb, totalAmount);
        add(sb, currency);
        add(sb, returnUrl);
        add(sb, returnMethod);
        if(null != getCustomer()) add(sb, getCustomer().toSign());
        add(sb, merchantData);
        add(sb, language);
        deleteLast(sb);
        return sb.toString();
    }

}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/LoanInitRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class LoanInitRequest : SignBaseRequest
{
    [JsonProperty("orderNo")] public string? OrderNo { get; set; }

    [JsonProperty("purchaseId")] public string? PurchaseId { get; set; }

    [JsonProperty("clientIp")] public string? ClientIp { get; set; }

    [JsonProperty("totalAmount")] public long? TotalAmount { get; set; }

    [JsonProperty("currency")] public string? Currency { get; set; }

    [JsonProperty("returnUrl")] public string? ReturnUrl { get; set; }

    [JsonProperty("returnMethod")] public string? ReturnMethod { get; set; }

    [JsonProperty("customer")] public LoanCustomer? Customer { get; set; }

    [JsonProperty("merchantData")] public string? MerchantData { get; set; }

    [JsonProperty("language")] public string? Language { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, OrderNo);
        Add(sb, PurchaseId);
        Add(sb, Dttm);
        Add(sb, ClientIp);
        Add(sb, TotalAmount);
        Add(sb, Currency);
        Add(sb, ReturnUrl);
        Add(sb, ReturnMethod);
        Add(sb, Customer?.ToSign());
        Add(sb, MerchantData);
        Add(sb, Language);
        DeleteLast(sb);
        return sb.ToString();
    }
}